docs/packages/ratings/index.js

```diff
@@ -0,0 +1,141 @@
+/* eslint-disable no-console, import/no-unresolved */
+const ApiBuilder = require('claudia-api-builder');
+const AWS = require('aws-sdk');
+const uuid = require('uuid/v4');
+
+const api = new ApiBuilder();
+const dynamoDb = new AWS.DynamoDB.DocumentClient();
+
+async function dbGet(request, id, page) {
+  const params = {
+    TableName: request.env.tableName,
+    Key: {
+      id,
+      page,
+    },
+    ConsistentRead: true,
+  };
+
+  try {
+    console.log('dbGet()', params);
+    const response = await dynamoDb.get(params).promise();
+    console.log({ response });
+    return response.Item;
+  } catch (error) {
+    console.error(error);
+    throw error;
+  }
+}
+
+async function dbPut(request, id, item = {}) {
+  const params = {
+    TableName: request.env.tableName,
+    Item: {
+      id,
+      page: request.body.page,
+      rating: request.body.rating,
+      comment: request.body.comment,
+      dateTime: new Date().toString(),
+    },
+  };
+
+  await Object.assign(params.Item, item);
+
+  try {
+    console.log('dbPut()', params);
+    const response = await dynamoDb.put(params).promise();
+    console.log({ response });
+    return params.Item;
+  } catch (error) {
+    console.error(error);
+    throw error;
+  }
+}
+
+async function dbQuery(request, id) {
+  const params = {
+    TableName: request.env.tableName,
+    KeyConditionExpression: 'id = :id',
+    ExpressionAttributeValues: {
+      ':id': id,
+    },
+    ConsistentRead: true,
+  };
+
+  try {
+    console.log('dbQuery()', params);
+    const response = await dynamoDb.query(params).promise();
+    console.log({ response });
+    return response.Items;
+  } catch (error) {
+    console.error(error);
+    throw error;
+  }
+}
+
+async function updateAverageRating(request, currentUserRating) {
```

Comment:
Does DynamoDB have queries for aggregate functions? I think we should prefer these. Otherwise we effectively double the queries. If the logic is bugged we have corrupted data unless we re-redeploy the fix.

Reply:
> Does DynamoDB have queries for aggregate functions?

Unfortunately not or I would indeed have used them. 

The way we're currently using the data, we don't actually need the average to be stored, but the function was written with star ratings in mind, with the ability to show the average rating for any particular page. It'll be handy to have it stored for back-end reporting, but it isn't critical.

The number of queries won't be a concern at this scale, so let's keep keep it for now. It'll be easy to prune if we decide later to drop it.

Reply:
Yeah if queries aren't limited then we might as well use them.